java/src/test/java/ai/rapids/cudf/ColumnVectorTest.java

```diff
@@ -2003,4 +2003,36 @@ void testStringTitlize() {
       assertColumnsAreEqual(expected, result);
     }
   }
+
+  @Test
+  void testNansToNulls() {
+    Float[] floats = new Float[]{1.2f, Float.POSITIVE_INFINITY,
+        Float.NEGATIVE_INFINITY, null, -Float.NaN, Float.NaN, Float.MAX_VALUE, Float.MIN_VALUE,
```

Comment:
`-Float.NaN` does not produce a negative NaN. Java will apply the negation operation to the NaN and end up with just another NaN.  You have to jump through some special hoops to get one. i.e. Create the bit representation of it in an int and call `Float.intBitsToFloat`.  There is a Double version with long as well.

Reply:
Thanks for pointing out the -Float.NaN. It is borderline embarrassing! 

I have used the intBitsToFloat in this same test. Not sure if you didn't look at it or are referring to something different from what I have used.